src/compiler/checker.ts

```diff
@@ -9329,7 +9393,7 @@ namespace ts {
             return getSignatureInstantiation(signature, getInferredTypes(context));
         }
 
-        function inferTypeArguments(node: CallLikeExpression, signature: Signature, args: Expression[], excludeArgument: boolean[], context: InferenceContext): void {
+        function inferTypeArguments(node: CallLikeExpression, signature: Signature, args: Expression[], excludeCallee: boolean, excludeArgument: boolean[], context: InferenceContext): void {

```

Comment:
I don't see why you need this new parameter. The callee should always be handled the same in type inference, i.e. if the signature has a `thisType`, infer from the type of the callee expression to the `thisType`.